pkg/sfu/pacer/no_queue.go

```diff
@@ -0,0 +1,82 @@
+package pacer
+
+import (
+	"sync"
+
+	"github.com/frostbyte73/core"
+	"github.com/gammazero/deque"
+	"github.com/livekit/protocol/logger"
+)
+
+type NoQueue struct {
+	*Base
+
+	logger logger.Logger
+
+	lock    sync.RWMutex
+	packets deque.Deque[Packet]
+	wake    chan struct{}
+	stop    core.Fuse
+}
+
+func NewNoQueue(logger logger.Logger) *NoQueue {
+	n := &NoQueue{
+		Base:   NewBase(logger),
+		logger: logger,
+		wake:   make(chan struct{}, 1),
+		stop:   core.NewFuse(),
+	}
+	n.packets.SetMinCapacity(9)
+
+	go n.sendWorker()
+	return n
+}
+
+func (n *NoQueue) Stop() {
+	n.stop.Once(func() {
+		close(n.wake)
+	})
+}
+
+func (n *NoQueue) Enqueue(p Packet) {
+	n.lock.Lock()
+	n.packets.PushBack(p)
+
+	notify := false
+	if n.packets.Len() == 1 {
+		notify = true
+	}
+	n.lock.Unlock()
+
+	if notify {
+		select {
+		case n.wake <- struct{}{}:
+		default:
+			n.logger.Warnw("wake channel full", nil)
+		}
+	}
+}
+
+func (n *NoQueue) sendWorker() {
+	for {
+		select {
+		case <-n.wake:
+			for {
+				n.lock.Lock()
+				if n.packets.Len() == 0 {
+					n.lock.Unlock()
+					break
+				}
+				p := n.packets.PopFront()
+				n.lock.Unlock()
+
+				n.Base.SendPacket(&p)
```

Comment:
All packets will go out in this go routine in this `Pacer` implementation. Idea is to use this with TWCC/send side bandwidth estimation. Note that all packets will go out from this go routine to ensure that the sequence number and time stamp we add for TWCC purposes do not get out-of-order before packet leaves the node. But, have to see if this creates a bottleneck or performance issue.

Reply:
It is also possible to try this with REMB (receive side BWE) to ensure packets are sent in order of time stamping to see if that improves bandwidth estimation, especially during probing.